monix-catnap-micrometer/src/main/scala/com/avast/sst/monix/catnap/micrometer/MicrometerCircuitBreakerMetricsModule.scala

```diff
@@ -0,0 +1,44 @@
+package com.avast.sst.monix.catnap.micrometer
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import cats.effect.Sync
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State.{Closed, HalfOpen, Open}
+import io.micrometer.core.instrument.MeterRegistry
+
+object MicrometerCircuitBreakerMetricsModule {
+
+  /** Makes [[com.avast.sst.monix.catnap.CircuitBreakerMetrics]] from [[io.micrometer.core.instrument.MeterRegistry]]. */
+  def make[F[_]: Sync](name: String, meterRegistry: MeterRegistry): F[CircuitBreakerMetrics[F]] = {
+    Sync[F].delay(new MicrometerCircuitBreakerMetrics(name, meterRegistry))
```

Comment:
why `Sync[F].delay`? is `MicrometerCircuitBreakerMetrics`'s constructor performing side effects?

Reply:
btw just something we ran into backend-common, somtimes it pays of to divorce F required to construct something from the F used by the actual thing, eg:

``` scala
  def make[F[_]: Applicative, G[_]: Sync](name: String, meterRegistry: MeterRegistry): F[CircuitBreakerMetrics[G]] = {
```

just something to keep in mind

Reply:
@sideeffffect To be honest I almost try to wrap everything in `F` just to be sure :) In this case would `new AtomicInteger` count as side effect?

@tomasherman Good point. I did not realize that but I think I'll redo all the modules (if possible) to allow that in some other PR.

Reply:
of course you're right @jakubjanecek , `new AtomicInteger(CircuitClosed)` needs to "be behind `Sync[F].delay`", bloody side effects hiding everywhere...

If we wan to make it super obvious, we could do something like this:
```scala
def make[F[_]: Sync](name: String, meterRegistry: MeterRegistry): F[CircuitBreakerMetrics[F]] =
  for {
    counter <- Sync[F].delay(new AtomicInteger(CircuitClosed))
    cbMetrics = new MicrometerCircuitBreakerMetrics(name, meterRegistry, counter)
  } yield cbMetrics

private val CircuitOpened = -1
private val CircuitHalfOpened = 0
private val CircuitClosed = 1 

private class MicrometerCircuitBreakerMetrics[F[_]: Sync](name: String, meterRegistry: MeterRegistry, counter: AtomicInteger) extends CircuitBreakerMetrics[F] {
  ...
}
```
just a suggestion...

Reply:
Yep, it look good.